app/lib/CaptureRecorder.js

```diff
@@ -41,7 +41,7 @@ function writeFile(filePath, data, cb) {
     }
   };
 
-  fs.writeFile(filePath, JSON.stringify(data, null, 2), cb);
+  fs.writeFileSync(filePath, JSON.stringify(data, null, 2));
```

Comment:
Reaching for `...Sync` is always a bummer.

Reply:
@ryanricard Agreed. We can find a better solution in future work. This is just a stopgap attempt to help mitigate the nondeterministic test failures on master. Luckily this isn't a production coed path with high throughput, and since we're streaming the response anyway, I think this work is backgrounded, so it shouldn't block individual responses, only slow down the event loop slightly.